src/main/kotlin/graphics/scenery/CurveGeometry.kt

```diff
@@ -0,0 +1,189 @@
+package graphics.scenery.proteins
+
+import cleargl.GLMatrix
+import cleargl.GLVector
+import com.jogamp.opengl.math.FloatUtil.makeRotationAxis
+import graphics.scenery.*
+import org.lwjgl.opengl.GL
+import java.nio.FloatBuffer
+import java.nio.IntBuffer
+import kotlin.math.acos
+import kotlin.math.sign
+/**
+ * Constructs a geometry along the calculates points of a Spline (in this case a Catmull Rom Spline).
+ * This class inherits from Node and HasGeometry
+ * The number n corresponds to the number of segments you wish to have between you control points.
+ * @author  Justin Buerger <burger@mpi-cbg.de>
+ */
+class CurveGeometry(curve: CatmullRomSpline, n: Int = 100): Node("CurveGeometry"), HasGeometry {
+    override val vertexSize = 3
+    override val texcoordSize = 2
+    override var geometryType = GeometryType.TRIANGLES
+
+    override var vertices: FloatBuffer = BufferUtils.allocateFloat(0)
+    override var normals: FloatBuffer = BufferUtils.allocateFloat(0)
+    override var texcoords: FloatBuffer = BufferUtils.allocateFloat(0)
+    override var indices: IntBuffer = BufferUtils.allocateInt(0)
+
+    private val cur = curve.CatMulRomChain(n)
+
+    /**
+     * This function renders the spline.
+     * @param [baseShape] It takes a lambda as a parameter, which is the shape of the
+     * curve. If you choose, for example, to have a square as a base shape, your spline will look like
+     * a banister. Please not that the base shape needs an equal number of points in each segments but it
+     * can very well vary in thickness.
+     */
+    fun drawSpline(baseShape: (() -> List<GLVector>)){
+        data class TranslationMatrix(val matrix: GLMatrix, val translation: GLVector)
+        val bases = ArrayList<TranslationMatrix>()
+        computeFrenetFrames().forEach { (t, n, b, tr) ->
+            if(n != null && b != null) {
+                val basisArray = ArrayList<Float>()
+                basisArray.add(n.x())
+                basisArray.add(b.x())
+                basisArray.add(t.x())
+                basisArray.add(0f)
+                basisArray.add(n.y())
+                basisArray.add(b.y())
+                basisArray.add(t.y())
+                basisArray.add(0f)
+                basisArray.add(n.z())
+                basisArray.add(b.z())
+                basisArray.add(t.z())
+                basisArray.add(0f)
```

Comment:
Instead of adding zeros here, you can stuff in the vector, making the additional GLVector superfluous (this is actually also how translations with homogeneous matrices work)